### ISSUE TITLE
Question: Tracker behavior with multiple selection

### ISSUE THREAD

**Issue by georgweiss** [ControlSystemStudio/phoebus#1702]
Currently the tracker box size is defined by the min and max coordinates of selected widgets. Any non-selected widgets within that box  cannot be selected in the drawing area, see screen shot.

Users are questioning this behavior with reference to other applications where multiple selection will render separate tracker boxes around the selected items only.

I'm not saying Phoebus does it wrong, but wonder about the rationale for the current implementation.

![Screenshot 2020-12-17 at 08 34 39](https://user-images.githubusercontent.com/35602960/102457704-754e8a00-4043-11eb-9509-57dc8c27bdf0.png)

**Comment by kasemir** [comment]
> Any non-selected widgets within that box cannot be selected in the drawing area, see screen shot.

You can. Command (Mac) respectively Ctrl-click (every other OS) within the tracker will add unselected widgets or remove already selected widgets.

The tracker uses the overall outline because 'resize' will affect that outline, and 'move' will move that outline. Besides, the most common operation is likely 'move', and this way you can move by dragging any part of the tracker, no need to hit within one of the potentially small selected items.

**Comment by kasemir** [comment]
Using Ctrl- resp. Command-click to add/remove from a selection seems pretty prevalent.
BOY did that, PowerPoint uses that, so most people are already familiar with the idea.

**Comment by georgweiss** [comment]
You're right, I did not know CMD/CTRL is the trick, neither did my user. My bad.

In any case, as you mention PowerPoint... Even though Phoebus mimics the behavior in terms of selection, move and resize, it is not obvious to what end the tracker is rendered the way it is. In particular as widgets included in the selection are marked with a reddish border.

**Comment by kasemir** [comment]
Selected widgets are marked with a red border to, well, indicate what's selected within the tracker outline, and support adding/removing selected widgets via cmd/ctrl click.

The tracker uses the overall outline to be easy to drag and resize. The 'handles' on the corners and centerlines are comparably large for the same reason. When instead placing one tracker on each selected widgets, these handles are necessarily smaller and then hard to click. You want to resize a widget, but miss the handler, instead click on the background and thus lose the selection.

**Comment by georgweiss** [comment]
OK, I see your point.
Thanks.

**Comment by kasemir** [comment]
The idea of basing the tracker on the outline may be unfamiliar to PowerPoint users, but it's actually quite common in CAD packages where fine-tuning the selection becomes important:

![sel](https://user-images.githubusercontent.com/1932421/102502819-11c95a00-404d-11eb-8b5b-45e763b8d556.PNG)

**Comment by georgweiss** [comment]
I agree comparisons need to account for additional applications besides PowerPoint. However, whether we like it or not, the Microsoft tools tend to set the standard for various aspects of software behavior.

**Comment by georgweiss** [comment]
Reopening as user reports issue.
In the attached screen shot three labels are placed on top of a background image. If the labels are selected in the indicated order, the last one cannot be selected as the background image is selected instead.
<img width="672" alt="Screenshot 2020-12-21 at 08 04 44" src="https://user-images.githubusercontent.com/35602960/102748841-41d16f80-4363-11eb-8eec-192c72f0a0c4.png">

**Comment by kasemir** [comment]
In this example it seems pretty straight forward to select the top, then the middle and finally the bottom label, which works fine.
Finally, there's the outline/tree view.

**Comment by georgweiss** [comment]
Yes, selecting in that order works. But top and bottom are selected first, middle cannot be selected.

**Comment by georgweiss** [comment]
@kasemir , can you please comment?

**Comment by kasemir** [comment]
All I can say is that deciding what to select in a 'click' when several widget overlap is hard.
I've described two ways to accomplish what you want:
Click in top, middle, down order, which you confirmed works.
Or use the tree view as an aid.
